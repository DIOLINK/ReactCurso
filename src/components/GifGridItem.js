import React from 'react';
import PropTypes from 'prop-types'; //impt
const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__bounce">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifGridItem.prototype = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
