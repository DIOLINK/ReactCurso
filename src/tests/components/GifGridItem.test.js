import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Prueba de GifGridItem.js', () => {
  const wrapper = shallow(
    <GifGridItem
      title="Title"
      url="https://giphy.com/gifs/studiosoriginals-wednesday-happy-morning-Cm9wKmKMUlRPvdoHgU"
    />
  );
  test('Debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Debe de tener un parrafo con el title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe('Title');
  });
  it('Img debe tener los prosp correctos', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(
      'https://giphy.com/gifs/studiosoriginals-wednesday-happy-morning-Cm9wKmKMUlRPvdoHgU'
    );
    expect(img.prop('alt')).toBe('Title');
  });
  test('Debe de tener un div con la clase "animate__animated animate__bounce"', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('animate__animated animate__bounce')).toBe(true);
  });
});
