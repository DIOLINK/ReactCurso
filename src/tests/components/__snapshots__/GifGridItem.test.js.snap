// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGridItem.js Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Title"
    src="https://giphy.com/gifs/studiosoriginals-wednesday-happy-morning-Cm9wKmKMUlRPvdoHgU"
  />
  <p>
    Title
  </p>
</div>
`;
